File: ollama/codigo-final/mandelbrot/codegemma-7b/cot/temp_0/Mandelbrot.java
import java.io.PrintStream;

public class Mandelbrot {

    private static final int MAX_ITERATIONS = 50;
    private static final double ESCAPE_THRESHOLD = 4.0;

    public static void main(String[] args) {
        int N = Integer.parseInt(args[0]);

        boolean[][] pixels = new boolean[N][N];

        for (int i = 0; i < N; i++) {
            for (int j = 0; j < N; j++) {
                pixels[i][j] = isInMandelbrot(mapPixelToComplex(i, j, N));
            }
        }

        printPbmP4(pixels, N);
    }

    private static boolean isInMandelbrot(Complex c) {
        Complex z = new Complex(0, 0);

        for (int i = 0; i < MAX_ITERATIONS; i++) {
            z = z.square().add(c);
            if (z.magnitude() > ESCAPE_THRESHOLD) {
                return false;
            }
        }

        return true;
    }

    private static Complex mapPixelToComplex(int i, int j, int N) {
        double x = -1.5 + (0.75 * i) / N;
        double y = -1.0 + (2.0 * j) / N;
        return new Complex(x, y);
    }

    private static void printPbmP4(boolean[][] pixels, int N) {
        PrintStream out = System.out;

        out.println("P4");
        out.println(N + " " + N);

        for (int i = 0; i < N; i++) {
            int bits = 0;
            for (int j = 0; j < N; j++) {
                bits |= (pixels[i][j] ? 1 : 0) << (N - j - 1);
                if (j == N - 1) {
                    out.print(bits);
                }
            }
            out.println();
        }
    }

    private static class Complex {
        private double real;
        private double imag;

        public Complex(double real, double imag) {
            this.real = real;
            this.imag = imag;
        }

        public Complex square() {
            return new Complex(real * real - imag * imag, 2 * real * imag);
        }

        public Complex add(Complex c) {
            return new Complex(real + c.real, imag + c.imag);
        }

        public double magnitude() {
            return Math.sqrt(real * real + imag * imag);
        }
    }
}

File: ollama/codigo-final/mandelbrot/codegemma-7b/cot/temp_0/mandelbrot.py
import sys

# Parse command-line argument
try:
    N = int(sys.argv[1])
except ValueError:
    sys.exit("Error: Invalid N value. Please provide a positive integer.")

# Constants
lim = 4.0
max_iterations = 50

# Coordinate mapping
def complex_plane_mapping(x, y):
    return complex(2.0 * x / N - 1.5, 2.0 * y / N - 1.0)

# Mandelbrot iteration
def is_in_mandelbrot(C):
    Z = 0 + 0j
    for _ in range(max_iterations):
        Z = Z**2 + C
        if abs(Z) > lim:
            return False
    return True

# Pack bits into bytes
def pack_bits(bits):
    packed_bytes = []
    byte = 0
    bit_count = 0
    for bit in bits:
        byte |= bit << (7 - bit_count)
        bit_count += 1
        if bit_count == 8:
            packed_bytes.append(byte)
            byte = 0
            bit_count = 0
    if bit_count > 0:
        packed_bytes.append(byte << (8 - bit_count))
    return packed_bytes

# Generate Mandelbrot set image
sys.stdout.buffer.write(b"P4\n" + b"%d %d\n" % (N, N))
bits = []
for y in range(N):
    for x in range(N):
        C = complex_plane_mapping(x, y)
        bits.append(int(is_in_mandelbrot(C)))

# Pack bits and write to output
packed_bytes = pack_bits(bits)
sys.stdout.buffer.write(b"".join(byte.to_bytes(1, "big") for byte in packed_bytes))
